File: Level2/구명보트.py
def solution(people, limit):
    answer = 0
    
    #정렬
    people.sort()
    
    #이분탐색처럼
    start = 0
    end = len(people) - 1
    
    #엇갈리기 전까지
    while (start <= end):
        #두명 데리고 가기
        if (people[start] + people[end] <= limit):
            start += 1
            end -= 1
        
        #한명만
        else:
            end -= 1
        
        #카운트
        answer +=1 
    
    #반환
    return answer

File: Level2/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    int size = people.size();

    int left = 0, right = size - 1;
    sort(people.begin(), people.end()); //오름 차순
    
    while(left <= right) {
        //한명만 데리고 가는 경우
        if (people[left] + people[right] > limit) 
            right--;
        
        //둘 다
        else {
            left++;
            right--;
        }
        answer++;
    }
    
    return answer;
}
